### PR TITLE
Include required ingressClassName for Ingress

### PR DESCRIPTION
**Pull request by SimonRupar**
Resolves: https://github.com/Genuitec/CodeTogether/issues/370

Includes required `spec.ingressClassName` as described in https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource

### CODE REVIEW

File: helm/templates/ingress.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+{{- if and .Values.ingress.className (semverCompare ">=1.19" .Capabilities.KubeVersion.GitVersion) }}
```
Comment (yurrriq) [inline helm/templates/ingress.yaml:24]: Does `semverCompare ">=1.19" .Capabilities.KubeVersion.GitVersion` work on EKS? I know they used to have funny versions.

File: helm/templates/ingress.yaml [hunk 2]
```diff
@@ -21,6 +21,9 @@ metadata:
...
+  ingressClassName: {{ .Values.ingress.className }}
```
Comment (yurrriq) [inline helm/templates/ingress.yaml:25]: I'd suggest we specify the empty string as the default in `values.yaml`:

```yaml
ingress:
  className: ""
```

**Review by yurrriq — CHANGES_REQUESTED** [review]
I made a couple inline suggestions. We should also bump the chart version.

File: helm/templates/ingress.yaml (hunk 1 above)
Comment (wgalanciak) [reply helm/templates/ingress.yaml:24]: Agree with @SimonRupar, let's keep it consistent with the rest of the chart. If there is an issue with EKS versioning, we should open a separate issue for it and fix it in all places.

**Comment by SimonRupar** [comment]
@wgalanciak, @yurrriq I have updated the changes based on PR.

File: helm/templates/ingress.yaml [hunk 3]
```diff
@@ -21,6 +16,9 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+{{- if and .Values.ingress.className (semverCompare ">=1.19" .Capabilities.KubeVersion.GitVersion) }}
```
Comment (wgalanciak) [inline helm/templates/ingress.yaml:19]: Just noticed, ingressClassName was introduced in 1.18 so this version should also be included in the condition.

**Review by wgalanciak — CHANGES_REQUESTED** [review]
Just one small change and we are ready for merging.

File: helm/templates/ingress.yaml (hunk 3 above)
Comment (SimonRupar) [reply helm/templates/ingress.yaml]: I have updated to 1.18 and squashed commits for traceability.

**Pull request merged**